geopandas/io/file.py

```diff
@@ -450,6 +485,37 @@ def _read_file_pyogrio(path_or_bytes, bbox=None, mask=None, rows=None, **kwargs)
     if kwargs.pop("ignore_geometry", False):
         kwargs["read_geometry"] = False
 
+    # translate `ignore_fields`/`include_fields` keyword for back compat with fiona
+    if "ignore_fields" in kwargs and "include_fields" in kwargs:
+        raise ValueError("Cannot specify both 'ignore_fields' and 'include_fields'")
+    elif "ignore_fields" in kwargs:
+        if kwargs.get("columns", None) is not None:
+            raise ValueError(
+                "Cannot specify both 'columns' and 'ignore_fields' keywords"
+            )
+        warnings.warn(
+            "It is recommended to use the 'columns' keyword to select the columns "
+            "you want instead of using 'ignore_fields' to deselect the columns you "
+            "do not want to read.",
+            stacklevel=2,
+        )
+        ignore_fields = kwargs.pop("ignore_fields")
+        fields = pyogrio.read_info(path_or_bytes)["fields"]
+        include_fields = [col for col in fields if col not in ignore_fields]
+        kwargs["columns"] = include_fields
+    elif "include_fields" in kwargs:
+        # translate `include_fields` keyword for back compat with fiona engine
+        if kwargs.get("columns", None) is not None:
+            raise ValueError(
+                "Cannot specify both 'columns' and 'include_fields' keywords"
+            )
+        warnings.warn(
+            "It is recommended to use the 'columns' keyword instead of the "
+            "'include_fields' keyword to select the columns to read.",
+            stacklevel=2,
+        )
```

Comment:
It could also be an option to more strongly discourage it, or even say it is deprecated (so we can remove this back compat at some point). In that case we should also warn for it in the fiona engine?